File: XCTestBootstrap/TestManager/FBTestManagerContext.m
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

#import "FBTestManagerContext.h"

@implementation FBTestManagerContext

- (instancetype)initWithSessionIdentifier:(NSUUID *)sessionIdentifier timeout:(NSTimeInterval)timeout testHostLaunchConfiguration:(FBApplicationLaunchConfiguration *)testHostLaunchConfiguration  testedApplicationAdditionalEnvironment:(nullable NSDictionary<NSString *, NSString *> *)testedApplicationAdditionalEnvironment testConfiguration:(FBTestConfiguration *)testConfiguration
{
  self = [super init];
  if (!self) {
    return nil;
  }

  _sessionIdentifier = sessionIdentifier;
  _timeout = timeout;
  _testHostLaunchConfiguration = testHostLaunchConfiguration;
  _testedApplicationAdditionalEnvironment = testedApplicationAdditionalEnvironment;
  _testConfiguration = testConfiguration;

  return self;
}

- (NSString *)description
{
  return [NSString stringWithFormat:
    @"Test Host %@ | Session ID %@ | Timeout %f",
    self.testHostLaunchConfiguration,
    self.sessionIdentifier.UUIDString,
    self.timeout
  ];
}

#pragma mark NSCopying

- (id)copyWithZone:(NSZone *)zone
{
  // Class is immutable.
  return self;
}

@end

File: XCTestBootstrap/TestManager/FBTestManagerContext.swift
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import FBControlCore
import Foundation

@objc public final class FBTestManagerContext: NSObject, NSCopying {

  @objc public let sessionIdentifier: UUID
  @objc public let timeout: TimeInterval
  @objc public let testHostLaunchConfiguration: FBApplicationLaunchConfiguration
  @objc public let testedApplicationAdditionalEnvironment: [String: String]
  @objc public let testConfiguration: FBTestConfiguration

  @objc public init(
    sessionIdentifier: UUID,
    timeout: TimeInterval,
    testHostLaunchConfiguration: FBApplicationLaunchConfiguration,
    testedApplicationAdditionalEnvironment: [String: String],
    testConfiguration: FBTestConfiguration
  ) {
    self.sessionIdentifier = sessionIdentifier
    self.timeout = timeout
    self.testHostLaunchConfiguration = testHostLaunchConfiguration
    self.testedApplicationAdditionalEnvironment = testedApplicationAdditionalEnvironment
    self.testConfiguration = testConfiguration
    super.init()
  }

  public override var description: String {
    return "Test Host \(testHostLaunchConfiguration) | Session ID \(sessionIdentifier.uuidString) | Timeout \(timeout)"
  }

  public func copy(with zone: NSZone? = nil) -> Any {
    return self
  }
}
